File: chain.R
library(dplyr)

df_diamonds <- read.csv("diamonds.csv")

df_diamonds %>%
  select(color, price) %>%
  group_by(color) %>%
  summarize(avg_price = mean(price)) %>%
  mutate(high_end = ifelse(avg_price > 5000, 1, 0))

File: chain.jl
using DataFramesMeta, DataFrames, CSV, Statistics

df_diamonds = CSV.File("diamonds.csv", header = 1, delim = ",") |> DataFrame

@linq df_diamonds |>
    select(:price, :color) |>
    by(:color, 
        avg_price = mean(:price), 
        high_end = mean(:price) > 5000 ? 1 : 0)
